librz/core/cmd/cmd_debug.c

```diff
@@ -4410,23 +4410,67 @@ RZ_IPI RzCmdStatus rz_cmd_debug_add_bp_handler(RzCore *core, int argc, const cha
 
 // dbl
 RZ_IPI RzCmdStatus rz_cmd_debug_list_bp_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
-	RzOutputMode mode = state->mode;
-	switch (mode) {
-	case RZ_OUTPUT_MODE_STANDARD:
-		rz_bp_list(core->dbg->bp, 0);
-		break;
-	case RZ_OUTPUT_MODE_RIZIN:
-		rz_bp_list(core->dbg->bp, 1);
-		break;
-	case RZ_OUTPUT_MODE_JSON:
-		rz_bp_list(core->dbg->bp, 'j');
-		break;
-	case RZ_OUTPUT_MODE_QUIET:
-		rz_bp_list(core->dbg->bp, -1);
-		break;
-	default:
-		rz_warn_if_reached();
-		break;
+	RzBreakpointItem *b;
+	RzListIter *iter;
+	RzOutputMode mode = RZ_OUTPUT_MODE_STANDARD;
+	PJ *pj = NULL;
+	if (state) {
+		mode = state->mode;
+		pj = state->d.pj;
+		rz_cmd_state_output_array_start(state);
+	}
+	rz_list_foreach (core->dbg->bp->bps, iter, b) {
+		switch (mode) {
+		case RZ_OUTPUT_MODE_STANDARD: {
+			rz_cons_printf("0x%08" PFMT64x " - 0x%08" PFMT64x
+						 " %d %c%c%c %s %s %s %s cmd=\"%s\" cond=\"%s\" "
+						 "name=\"%s\" module=\"%s\"\n",
+				b->addr, b->addr + b->size, b->size,
+				((b->perm & RZ_BP_PROT_READ) | (b->perm & RZ_BP_PROT_ACCESS)) ? 'r' : '-',
+				((b->perm & RZ_BP_PROT_WRITE) | (b->perm & RZ_BP_PROT_ACCESS)) ? 'w' : '-',
+				(b->perm & RZ_BP_PROT_EXEC) ? 'x' : '-',
+				b->hw ? "hw" : "sw",
+				b->trace ? "trace" : "break",
+				b->enabled ? "enabled" : "disabled",
+				rz_bp_is_valid(core->dbg->bp, b) ? "valid" : "invalid",
+				rz_str_get(b->data),
+				rz_str_get(b->cond),
+				rz_str_get(b->name),
+				rz_str_get(b->module_name));
+			break;
+		}
```

Comment:
@ret2libc @XVilka maybe using the table api is better?

Reply:
I would love to have it, yes, but for the sake of this single PR i'm ok also as is.

Reply:
I will add tables. Looks nice to me.

Reply:
This was not resolved yet, was it? @DMaroo are you going to do it in this PR?

Reply:
Yes, I am.

Reply:
I tried implementing it using `RzTable`, but I now do not feel it is worth it. Anyways, we won't be able to use `RzTable` for `RZ_OUTPUT_MODE_JSON` since that would require us the JSON object (which the current `RzTable` does not provide). Therefore, the only use case for the table is `RZ_OUTPUT_MODE_STANDARD`, which already uses a single `rz_cons_printf` line. Using tables would require me to create a `RzList` of all the attributes and then create a row from it and then insert it to the table. I don't think the extra complexity is necessary here. Sorry for delaying, I don't think we need tables. This is resolved from my end.

Reply:
I'm ok to add it later, but the table does not require you to have rzlist.
You can just use rz_table_add_rowf, e.g. `rz_table_add_rowf(state->d.t, "sssss", str, plugin->license, plugin->name, plugin->uris, plugin->desc);`. In general, it allows you more control on the output and it provides the headers, which are always nice to have as a user immediately understands what the output means.

Reply:
Well, then I'll just add it.